File: timesaving.c
#include <stdio.h>
#include <time.h>

// Função para obter a data e a hora atuais
void getCurrentDateTime(char *dateTime) {
    time_t rawtime;
    struct tm *info;

    time(&rawtime);
    info = localtime(&rawtime);

    strftime(dateTime, 80, "%Y-%m-%d %H:%M:%S", info);
}

// Função para registrar a data e hora em um arquivo de log
void logDateTime(const char *dateTime) {
    FILE *file;
    file = fopen("log.txt", "a");

    if (file != NULL) {
        fprintf(file, "%s\n", dateTime);
        fclose(file);
    } else {
        printf("Erro ao abrir o arquivo de log.\n");
    }
}

int main() {
    char dateTime[80];

    // Obter a data e a hora atuais
    getCurrentDateTime(dateTime);

    // Registrar a data e a hora no log
    logDateTime(dateTime);

    // Retornar o log
    printf("Log:\n");

    FILE *file;
    file = fopen("log.txt", "r");

    if (file != NULL) {
        char c;
        while ((c = getc(file)) != EOF) {
            putchar(c);
        }
        fclose(file);
    } else {
        printf("Erro ao abrir o arquivo de log.\n");
    }

    return 0;
}

File: timesaving.py
import datetime

# Função para obter a data e a hora atuais
def get_current_datetime():
    return datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")

# Função para registrar a data e hora em um arquivo de log
def log_datetime(date_time):
    with open("log.txt", "a") as file:
        file.write(date_time + "\n")

# Função para ler e retornar o conteúdo do log
def read_log():
    try:
        with open("log.txt", "r") as file:
            return file.read()
    except FileNotFoundError:
        return "Arquivo de log não encontrado."

def main():
    # Obter a data e a hora atuais
    current_date_time = get_current_datetime()

    # Registrar a data e a hora no log
    log_datetime(current_date_time)

    # Retornar o conteúdo do log
    print("Log:")
    print(read_log())

if __name__ == "__main__":
    main()
